### ISSUE TITLE
web: Add better error messages for 404 and CORS issues

### ISSUE THREAD

**Issue by Herschel** [ruffle-rs/ruffle#1824]
We should probably display the error dialog with an informative message for cases such as:
 * ruffle.wasm fails to load due to 404
 * SWF fails to load due to 404/invalid URL
 * SWF fails to load due to CORS issues

**Comment by GreenReaper** [comment]
Yeah, Ruffle breaks on our existing CDN setup since loading an SWF file was not previously an operation subject to CORS.
This is relatively simple to debug in the console, but not indicated in the UI, even though it detects an asynchronous error.
NoScript *also* breaks loading (until allowed), showing as "Blocked by Extension" rather than "CORS Missing Allow Origin".

The wiki could either be updated with minimal CORS configuration for nginx and Apache ([similar to here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSMissingAllowOrigin)), or just link to it.

P.S. [The current mime-type error link](https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle#configure-wasm-mime-type) doesn't scroll to the right anchor because "WASM" has changed to "WebAssembly".
Not sure if you want to change that in the code or on the wiki.

**Comment by relrelb** [comment]
Fixed by #2622?

**Comment by GreenReaper** [comment]
I think it's fixed; any specific ideas for improvements to it could come in another issue (and probably have already).
